api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto

```diff
@@ -64,4 +64,29 @@ message RateLimit {
   // success.
   config.ratelimit.v3.RateLimitServiceConfig rate_limit_service = 7
       [(validate.rules).message = {required: true}];
+
+  // If set to true enables emitting X-RateLimit-* headers by the filter:
+  //
+  // * ``X-RateLimit-Limit`` - indicates the request-quota associated to the
+  //   client in the current time-window followed by the description of the
+  //   quota policy. The values are returned by the rate limiting service in
+  //   :ref:`current_limit<envoy_v3_api_field_service.ratelimit.v3.RateLimitResponse.DescriptorStatus.current_limit>`
+  //   field. Example: `10, 10;w=;name="per-ip", 1000;window=3600`.
+  // * ``X-RateLimit-Remaining`` - indicates the remaining requests in the
+  //   current time-window. The values are returned by the rate limiting service
+  //   in :ref:`limit_remaining<envoy_v3_api_field_service.ratelimit.v3.RateLimitResponse.DescriptorStatus.limit_remaining>`
+  //   field.
+  // * ``X-RateLimit-Reset`` - indicates the number of seconds until reset of
+  //   the current time-window. The values are returned by the rate limiting service
+  //   in :ref:`duration_until_reset<envoy_v3_api_field_service.ratelimit.v3.RateLimitResponse.DescriptorStatus.duration_until_reset>`
+  //   field.
+  //
+  // In case rate limiting policy specifies more then one time window, the values
+  // above represent the window that is closest to reaching its limit.
+  //
+  // For more information about the headers specification see the `draft RFC
+  // <https://tools.ietf.org/id/draft-polli-ratelimit-headers-02.html>`_
+  //
+  // Defaults to false.
+  bool enable_x_ratelimit_headers = 8;
```

Comment:
Sorry one last thought here: given this is a draft RFC, we may ultimately have to support multiple versions. WDYT about an enum in which the default value is off, then the other value specifies the RFC version? This would allow us to support other versions in the future and deprecate versions eventually?